src/xrpld/app/tx/detail/Batch.cpp

```diff
@@ -145,13 +216,64 @@ Batch::preflight(PreflightContext const& ctx)
         if (auto const preflightResult = ripple::preflight(
                 ctx.app, ctx.rules, parentBatchId, stx, tapBATCH, ctx.j);
             preflightResult.ter != tesSUCCESS)
+        {
+            JLOG(ctx.j.trace())
+                << "BatchTrace[" << parentBatchId << "]: "
+                << "inner txn preflight failed: " << preflightResult.ter << " "
+                << "txID: " << hash;
+            return temINVALID_INNER_BATCH;
+        }
+
+        // Check that the fee is zero
+        if (auto const fee = stx.getFieldAmount(sfFee);
+            !fee.native() || fee.xrp() != beast::zero)
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn must have a fee of 0. "
+                                << "txID: " << hash;
+            return temINVALID_INNER_BATCH;
+        }
+
+        // Check that Sequence and TicketSequence are not both present
+        if (stx.isFieldPresent(sfTicketSequence) &&
+            stx.getFieldU32(sfSequence) != 0)
```

Comment:
should maybe also check if both fields are missing

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5060/commits/64483176cadaaef4926fb4533e4957f9e7511135